BUILD.md

```diff
@@ -49,3 +49,13 @@ Developers who can work a command line may be interested in using [Visual Studio
 * `docker build --rm -t northstar-build-fedora .`
 * `docker run --rm -it -e CC=cl -e CXX=cl --mount type=bind,source="$(pwd)",destination=/build northstar-build-fedora cmake . -DCMAKE_BUILD_TYPE=Release -DCMAKE_SYSTEM_NAME=Windows -G "Ninja"`
 * `docker run --rm -it -e CC=cl -e CXX=cl --mount type=bind,source="$(pwd)",destination=/build northstar-build-fedora cmake --build .`
+
+#### Podman
+
+When using [`podman`](https://podman.io/) instead of Docker on an SELinux enabled distro, make sure to add the `Z` flag when mounting the directory to keep SELinux happy.
```

Comment:
> "keep SELinux happy"

tbh, I don't understand SELinux well enough yet to explain why this works. As far as I understood, SELinux uses labels to mark whether a process is allowed to access a certain directory. Without the `Z` the label is not applied to the directory and as such access is denied?

Maybe someone can elaborate on this, otherwise I'd just merge as is ^^

Reply:
SELinux labels are used to define the context of and around a file to ensure specific processes (e.g. nginx) can only read anything context relevant (e.g. not `user_home_t` aka anything in a user home) even if they permissions are wrongly defined.

This is troublesome when you use containers that mount or bind outside contents because it may be labeled in a way that makes SELinux prevent us from accessing it

The `z` and `Z` options exist in podman and docker to make it add relabel the content in a way that makes it readable but gives them a hint on how its suppose to be used (`z` being shared, `Z` being private).

These kind of problems with SELinux are wide spread and why SELinux is commonly advised to be disabled on Desktop Systems or outright replaced in some distros (e.g. Nobara).

Question is, why does this happen on Podman but not Docker?
while Podman aims to be compatible, they are far from the same and there are many subtle differences in their behavior. Could be anything from podman's rootless nature to better selinux integration in docker.

Reply:
Thanks for the thorough explanation <3

> Question is, why does this happen on Podman but not Docker?
> while Podman aims to be compatible, they are far from the same and there are many subtle differences in their behavior. Could be anything from podman's rootless nature to better selinux integration in docker.

Pretty sure it's either Docker running as root or podman being better integrated with SELinux that results in this.

(Docker running as root and creating root owned build caches and artifacts is the primary reason I looked into building with Podman ^^)

Reply:
Sorta addressed with 3936f7999be4d827641939c3111785c6be3381e9 I guess